src/app/util/ember-compatibility-functions.cpp

```diff
@@ -281,6 +288,16 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
         ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), int16_data));
         break;
     }
+    case ZCL_INT24S_ATTRIBUTE_TYPE: // Signed 24-bit integer
+    {
+        int32_t int32_data;
+        int32_data = emberAfGetInt24s(attributeData, 0, 3);
+        // Trigger the sign extension
+        int32_data <<= 8;
+        int32_data >>= 8;
```

Comment:
Shouldn't that happen inside `emberAfGetInt24s`?  It sounds like that's not implemented correctly....

And again, the endianness question.

Reply:
Yes, I think it should happen inside the method, but it does not - and I was currently forced to use this special construction "to grab the sign". Perhaps it should go into the `emberAfGetUnt24s()` method is self?

Reply:
Ideally yes.  Followup for that is fine.  But in general, I suspect we don't want to use `emberAfGetUnt24s` here anyway because of the endianness problem: it always reads little-endian data.